### PR TITLE
[Bug] fix some error on build.sh && fix build fail with clang on runtime_profile

### LINKED ISSUE #8747: [Bug] fix some error on build.sh && fix build fail with clang on runtime_profile
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

1. fix some error on build.sh 
2. fix build fail with clang on runtime_profile, and do some modernize refactor

### What You Expected?

fix it

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

Issue Number: close #8747

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by yiguolei — APPROVED** [review]
LGTM

File: build.sh [hunk 1]
```diff
@@ -135,7 +135,7 @@ if [ $# == 1 ] ; then
     BUILD_FE=1
     BUILD_BE=1
     BUILD_BROKER=1
-    BUILD_META_TOOL=1
```
Comment (caiconghui) [inline build.sh:138]: why change value from 1 to on here?

Comment (BiteTheDDDDt) [reply build.sh:138]: > why change value from 1 to on here?

to apapt cmakelist
```
if (BUILD_META_TOOL AND BUILD_META_TOOL STREQUAL "ON")
    add_subdirectory(${SRC_DIR}/tools)
```

Comment (caiconghui) [reply build.sh:138]: I think it is better to keep one style for build option，how about change cmakelist by modfiy “on” to ”1“？

**Pull request merged**